[NCCL] Expose NCCL-specific collective options to Python

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43675 Test for NCCL Stream Overwrite
* **#41356 [NCCL] Expose NCCL-specific collective options to Python**
* #41276 [NCCL] Adding CUDA Stream Overwrite Options for Allgather, Reduce, ReduceScatter, and Broadcast
* #41275 [NCCL] Overwrite NCCL Stream in AllReduce

This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

osalpekar committed:
[NCCL] Expose NCCL-specific collective options to Python

This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

jiayisuse commented:
Seem CI test fails. Let's take a look at the failures like test_broadcast_coalesced_nccl

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]

osalpekar committed:
Update on "[NCCL] Expose NCCL-specific collective options to Python"


This PR adds the new NCCL-specific collective options structs to the
pybind definition file (c10d/init.cpp) so the structs are exposed to Python.

Differential Revision: [D22508373](https://our.internmc.facebook.com/intern/diff/D22508373/)

[ghstack-poisoned]
